### PR TITLE
Convert error strings to objects - Closes #3189 

### LINKED ISSUE #3189: All of the Error needs to converted to error object
### Expected behavior
String should not be used as an error

### Actual behavior
String is used as error through out the code

### Which version(s) does this affect? (Environment, OS, etc...)
1.7-

### PR DESCRIPTION
**Pull request by mitsujutsu**
### What was the problem?

Application was inconsistently throwing strings instead of Error objects in some areas of the code.

### How did I fix it?

Converted thrown strings to Error Objects and related changes.

API related error changes to be handled in separate PR. 

### How to test it?

npm run mocha:unit

### Review checklist

* The PR resolves #3189 
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: framework/src/modules/chain/submodules/accounts.js [hunk 1]
```diff
@@ -201,7 +201,7 @@ Accounts.prototype.setAccountAndGet = function(data, cb, tx) {
 
 	return taskPromise
 		.then(taskPromiseData => setImmediate(cb, null, taskPromiseData))
-		.catch(taskPromiseErr => setImmediate(cb, taskPromiseErr));
```
Comment (SargeKhan) [inline framework/src/modules/chain/submodules/accounts.js]: Won't the error here already be an object?

File: framework/src/modules/chain/submodules/blocks/chain.js [hunk 2]
```diff
@@ -488,7 +488,8 @@ __private.loadSecondLastBlockStep = function(secondLastBlockId, tx) {
 			{ id: secondLastBlockId },
 			(err, blocks) => {
 				if (err || !blocks.length) {
-					library.logger.error('Failed to get loadBlocksPart', err);
+					const error = Array.isArray(err) && err.length > 0 ? err[0] : err;
+					library.logger.error('Failed to get loadBlocksPart', error);
 					return setImmediate(reject, err || 'previousBlock is null');
```
Comment (SargeKhan) [inline framework/src/modules/chain/submodules/blocks/chain.js]: 'previousBlock is null' should be wrapped inside an error object as well. Right?

File: framework/src/modules/chain/logic/block.js [hunk 3]
```diff
@@ -251,9 +253,7 @@ class Block {
 				throw invalidTransactionResponse.errors;
 			}
 		} catch (errors) {
-			const error =
-				Array.isArray(errors) && errors.length > 0 ? errors[0] : errors;
-			throw error;
+			throw errors;
```
Comment (pablitovicente) [inline framework/src/modules/chain/logic/block.js]: Are the items contained in `errors` Error instances?

Comment (mitsujutsu) [reply framework/src/modules/chain/logic/block.js]: Yes or more specifically `TransactionError` from elements

**Comment by limiaspasdaniel** [comment]
Sorry for jumping in here. 

Please bear in mind that **Error** objects are not correctly logged in our current application state. 

For example: https://github.com/LiskHQ/lisk/blob/949b41e80eb30992385f90b5b59bf7006b388a73/framework/src/controller/controller.js#L294
The controller in this case needs to be prepared to receive objects instead of plain `strings`.
You can reproduce this problem by for example running an instance of the application and starting another one with the same port. You'll see how the following log shows:
`Caused error during cleanup - {}` demonstrating what I've stated above.
I don't know if this is included in this PR's scope but definitely needs to be addressed perhaps in another issue.

File: framework/src/modules/chain/logic/block.js [hunk 4]
```diff
@@ -251,9 +253,7 @@ class Block {
 				throw invalidTransactionResponse.errors;
 			}
 		} catch (errors) {
```
Comment (shuse2) [inline framework/src/modules/chain/logic/block.js]: no need to catch here anymore

**Comment by mitsujutsu** [comment]
> Is it only 18 files? I feel like there should be more...

There is few more in transport but I will change those for the next issue: #3152 as it is related.

**Comment by mitsujutsu** [comment]
> Is it only 18 files? I feel like there should be more...

That is most. There are few left in transport but I will update it in next API Error PR since it is related.

> A quick search in the repository reveals that there are still some places where we are calling callback with error strings. For-example,
> 
> ```
> framework/test/mocha/unit/modules/chain/scripts/json_history.js|182 col 7| cb('Error occurred during change.');
> framework/test/mocha/unit/modules/chain/submodules/multisignatures.js|285 col 6| cb('Err', null);
> framework/test/mocha/unit/modules/chain/submodules/delegates.js|890 col 6| cb('generateDelegateList error');
> framework/test/mocha/common/utils/wait_for.js|65 col 12| return cb('Failed to load blockchain');
> framework/test/mocha/common/utils/wait_for.js|81 col 12| return cb('Server is not responding');
> framework/test/mocha/functional/ws/transport/handshake.js|76 col 11| return cb('Socket should not connect but it did');
> framework/src/modules/chain/submodules/loader.js|846 col 22| return waterCb('Failed to get random peer from network');
> framework/src/modules/chain/submodules/loader.js|887 col 22| return waterCb(`Failed to load blocks from: ${peer.string}`);
> ```
> 
> Please double check the codebase for any missed cases.

The ones left in `src` are for loader.js `waterCb`. It is an `async waterfall` in which the error message is logged at the end of the waterfall, therefore we can leave as a message. I will update the unit ones for consistency but I think they pass as is.

**Pull request merged**